File: 70. Climbing Stairs.java
class Solution {
    public int climbStairs(int n) {
        int[] mem = new int[n];

        if(n == 1){
            return 1;
        }
    
        mem[0] = 1;
        mem[1] = 2;
        int i = 2;

        while(i < n){
            mem[i] = mem[i-1] + mem[i-2];
            i++;
        }

        return mem[n-1];
    }
}

File: 70. Climbing Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n==1:
            return 1
        steparr = [0]*n
        steparr[0] = 1
        steparr[1] = 2
        for i in range(2,n):
            steparr[i] = steparr[i-1] + steparr[i-2]
        return steparr[n-1]



        
